packages/venia-ui/lib/components/Wishlist/WishlistDialog/NewWishlistForm/newWishlistForm.js

```diff
@@ -0,0 +1,130 @@
+import React, { Fragment } from 'react';
+import { useIntl } from 'react-intl';
+import { Relevant } from 'informed';
+
+import { useNewWishlistForm } from '@magento/peregrine/lib/talons/Wishlist/WishlistDialog/NewWishlistForm/useNewWishlistForm';
+
+import Button from '@magento/venia-ui/lib/components/Button';
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+import Field from '@magento/venia-ui/lib/components/Field';
+import TextInput from '@magento/venia-ui/lib/components/TextInput';
+import RadioGroup from '@magento/venia-ui/lib/components/RadioGroup';
+import { isRequired } from '@magento/venia-ui/lib/util/formValidators';
+
+import defaultClasses from './newWishlistForm.css';
+import FormError from '../../../FormError';
+
+const NewWishlistForm = props => {
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const { formatMessage } = useIntl();
+    const talonProps = useNewWishlistForm({
+        onCreateList: props.onCreateList,
+        isAddLoading: props.isAddLoading
+    });
+
+    const {
+        formErrors,
+        handleCancel,
+        handleNewListClick,
+        handleSave,
+        isOpen,
+        isSaveDisabled
+    } = talonProps;
+
+    const privateRadioText = formatMessage({
+        id: 'newWishlistForm.privateRadio',
+        defaultMessage: 'Private'
+    });
+
+    const publicRadioText = formatMessage({
+        id: 'newWishlistForm.publicRadio',
+        defaultMessage: 'Public'
+    });
+
+    const createButtonText = formatMessage({
+        id: 'newWishlistForm.createButton',
+        defaultMessage: '+ Create a new list'
+    });
+
+    const cancelButtonText = formatMessage({
+        id: 'newWishlistForm.createButton',
+        defaultMessage: 'Cancel'
+    });
+
+    const saveButtonText = formatMessage({
+        id: 'newWishlistForm.saveButton',
+        defaultMessage: 'Save'
+    });
+
+    const maybeForm = isOpen ? (
+        <Fragment>
```

Comment:
Should this be its own component?

Reply:
Handled - Moved the button up to the dialog, now this is just the form itself.